spring-session/src/main/java/org/springframework/session/data/cassandra/CassandraSessionRepository.java

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright 2014-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.session.data.cassandra;
+
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import com.datastax.driver.core.BatchStatement;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Statement;
+import com.datastax.driver.core.querybuilder.Insert;
+import com.datastax.driver.core.querybuilder.QueryBuilder;
+import com.datastax.driver.core.querybuilder.Select;
+import com.google.common.reflect.TypeToken;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import org.springframework.data.cassandra.core.CassandraTemplate;
+import org.springframework.session.ExpiringSession;
+import org.springframework.session.FindByIndexNameSessionRepository;
+import org.springframework.session.MapSession;
+import org.springframework.session.data.cassandra.conversion.SessionAttributeDeserializer;
+import org.springframework.session.data.cassandra.conversion.SessionAttributeSerializer;
+import org.springframework.session.jdbc.JdbcOperationsSessionRepository;
+
+
+/**
+ * TODO doc this.
+ * @author Andrew Fitzgerald
+ */
+public class CassandraSessionRepository implements FindByIndexNameSessionRepository<CassandraSessionRepository.CassandraHttpSession> {
+
+	private static final Logger log = LoggerFactory.getLogger(CassandraSessionRepository.class);
+	private static final JdbcOperationsSessionRepository.PrincipalNameResolver PRINCIPAL_NAME_RESOLVER = new JdbcOperationsSessionRepository.PrincipalNameResolver();
```

Comment:
Cool, I had a mental TODO of extracting that to a common class, will just copy-paste for now.